packages/@aws-cdk/aws-s3-notifications/test/lambda/lambda.test.ts

```diff
@@ -29,20 +27,20 @@ test('add notifications to multiple functions', () => {
   bucket.addEventNotification(s3.EventType.OBJECT_CREATED, lambdaDestination2, { prefix: 'v2/' });
 
   // expecting notification configuration to have both events
-  expect(stack).toHaveResourceLike('Custom::S3BucketNotifications', {
+  Template.fromStack(stack).hasResourceProperties('Custom::S3BucketNotifications', {
     NotificationConfiguration: {
       LambdaFunctionConfigurations: [
-        { Filter: { Key: { FilterRules: [{ Name: 'prefix', Value: 'v1/' }] } } },
-        { Filter: { Key: { FilterRules: [{ Name: 'prefix', Value: 'v2/' }] } } },
+        Match.objectLike({ Filter: { Key: { FilterRules: [{ Name: 'prefix', Value: 'v1/' }] } } }),
```

Comment:
Unfortunate that the partial matching behavior that used to be inherited from `toHaveResourceLike` is dropped when matching an array (or at least, that's what I'm assuming is happening).

Is that on purpose? And if it is on purpose, what is the rationale? Is that documented?